interp/scope.go

```diff
@@ -130,6 +132,20 @@ func (s *scope) lookup(ident string) (*symbol, int, bool) {
 	return nil, 0, false
 }
 
+// lookdown searches for a symbol in the current scope and included ones, recursively.
+// It returns the first found symbol and true, or nil and false.
```

Comment:
if the returned symbol is always nil when not found, then its "nilness" can always be used as the sentinel in the caller, and the boolean is not actually needed. Not a big deal though.

Reply:
hmm, but I suppose have a (val,ok) form allows the caller to do "if _,ok := sc.lookdown", which I imagine may allow some memory optimizations when the compiler knows the first returned element is not needed and can be skipped. so maybe it's always better to have a comma-ok form...

Reply:
I am not entirely sure that a symbol with a `nil` value is always invalid, and also I'd like to keep API consistency with symetrical `lookup` method.